### PR TITLE
add a dummy builder, for syntax checkers

### PR DESCRIPTION
**Pull request by bukzor**
Sometimes we'd like to just do validation of our rst files, without writing to disk.

See also https://github.com/scrooloose/syntastic/pull/1454

### CODE REVIEW

**Comment by bukzor** [comment]
Other possible names:
- null -- the null builder.
- syntax -- the builder which only validates syntax.

**Comment by bukzor** [comment]
ping: six months

**Comment by shimizukawa** [comment]
Sorry for late reply.
LGTM except below:
- builder name: `null` is better to me.
- simple test is required. please add builder name to https://github.com/sphinx-doc/sphinx/blob/master/tests/test_build.py#L69

**Comment by tk0miya** [comment]
:+1: but, `null` is not either formats or behaviors.
So I think `syntaxcheck` is better. It is more easy to understand what is it.

**Comment by lehmannro** [comment]
I wonder if calling it `syntaxcheck` is a little ambitious.  In terms of builders, it's really more of a dummy/null operation.

In case of doubt, I'd give the privilege of naming it to the original author instead of bikeshedding.

**Comment by shimizukawa** [comment]
> In case of doubt, I'd give the privilege of naming it to the original author instead of bikeshedding.

+1

**Comment by tk0miya** [comment]
> I wonder if calling it syntaxcheck is a little ambitious. In terms of builders, it's really more of a dummy/null operation.

To be cleared, I think `null` is good for name of builder. but, at the same time, `null` is not better for make target. In other words, I think and `make syntaxcheck` is better.

> In case of doubt, I'd give the privilege of naming it to the original author instead of bikeshedding.

I agreed that.
@bukzor Could you name it please?

**Comment by tk0miya** [comment]
ping. any progress?

File: doc/builders.rst [hunk 1]
```diff
@@ -341,6 +341,18 @@ for details.
...
 
+.. module:: sphinx.builders.dummy
+.. class:: DummyBuilder
+
+   This builder produces no output.  The input is only parsed and checked for
+   consistency.  This is useful for linting purposes.
+
+   .. autoattribute:: name
+
+   .. autoattribute:: supported_image_types
```
Comment (lehmannro) [inline doc/builders.rst:352]: Does it really make sense to document image types here?

**Comment by lehmannro** [comment]
Apart from my small docs nit (which I can also fix myself later) I'm good to merge this _as-is._

File: doc/builders.rst (hunk 1 above)
Comment (shimizukawa) [inline doc/builders.rst:352]: I think so. Sometime I want to know which formats are supported by a builder.
In this case, the documentation will indicate the dummy builder supports no image format explicitly.

Comment (tk0miya) [inline doc/builders.rst:352]: In the page, almost of builders have `supported_image_types` entry if the builder supports no image format
http://www.sphinx-doc.org/en/stable/builders.html#sphinx.builders.text.TextBuilder

I'd rather adding `format` attribute than removing it.

**Comment by tk0miya** [comment]
Ok, I'll rename this to `NullBuilder` and merge it within this few days.

**Comment by lehmannro** [comment]
I'd say merge as-is;  the author called it _dummy_ builder, let's keep it that way if we don't hear otherwise. :-)

**Comment by tk0miya** [comment]
OK, I will merge it without renaming :-)

**Comment by tk0miya** [comment]
Merged (of course without renaming :-). Thank you for cotnribution!

**Pull request merged**